tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -103,7 +103,14 @@ def test_geotransform_and_friends(self):
         self.assertEqual(self.rs.skew.y, 0)
         # Create in-memory rasters and change gtvalues
         rsmem = GDALRaster(JSON_RASTER)
+        rsmem.geotransform = [0.0, 1.0, 2.0, 3.0, 4.0, 5.0]
         rsmem.geotransform = range(6)
+        with self.assertRaisesMessage(ValueError, 'Geotransform must consist of 6 numeric values.'):
+            rsmem.geotransform = [1, 2]
+        with self.assertRaisesMessage(ValueError, 'Geotransform must consist of 6 numeric values.'):
+            rsmem.geotransform = [1, 2, 3, 4, 5, 'foo']
+        with self.assertRaisesMessage(ValueError, 'Geotransform must consist of 6 numeric values.'):
+            rsmem.geotransform = [1, 2, 3, 4, 5, 6, 'foo']
```

Comment:
sure, I forget about subtest because I use pytest most of the time. normally I'd even make these separate tests